modules/javafx.controls/src/main/java/javafx/scene/control/skin/SpinnerSkin.java

```diff
@@ -249,6 +250,12 @@ public void executeAccessibleAction(AccessibleAction action, Object... parameter
                 return null;
             }
         }));
+
+        getSkinnable().sceneProperty().addListener((observable, oldValue, newValue) -> {
```

Comment:
this will permanently add a listener on the control's scene property, creating a memory leak when skins are replaced.
You probably want to use the newly introduced ListenerHelper (there is an instance of it right there in the constructor):
```
        lh.addChangeListener(control.sceneProperty(), (x) -> {
            behavior.stopSpinning();
        });
```

doing this will decouple the listener on Skin.dispose()

Reply:
Yes, and the failing unit tests show exactly this leak.

Reply:
Made changes to use `ListenereHelper` instead of permanently adding listener on control's `sceneProperty`. Please check.